R/module_teal.R

```diff
@@ -136,6 +138,29 @@ srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices()) {
       title = "SessionInfo"
     )
 
+    output$lockFile <- downloadHandler(
+      filename = function() {
+        "renv.lock"
+      },
+      content = function(file) {
+        user_lockfile <- getOption("teal.renv.lockfile", "")
+        if (!file.exists(user_lockfile)) {
+          teal_lockfile <- getOption("teal.internal.renv.lockfile")
+          iter <- 1
+          while (!file.exists(teal_lockfile) && iter <= 100) {
+            logger::log_trace("lockfile not created yet, retrying...")
```

Comment:
don't you have to specify logger namespace?

```suggestion
            logger::log_trace("lockfile not created yet, retrying...", namespace = "teal")
```

( you see? yet another rule that is not enforced and we rely that devs won't forget :P )

Reply:
@pawelru No I don't think we need that. None other `logger::log_trace` has `namespace` specified in the whole `teal` package. I think the namespace is handled by

```r
teal.logger::register_logger("teal")
teal.logger::register_handlers("teal")
```

that is called in `R/zzz.R` during `.onLoad`

Reply:
`teal.logger::register_logger("teal")` - this is _creating_ a logger namespace
`teal.logger::register_handlers("teal")` - this will pass all the warnings / errors to that logger namespace

This still feels incorrect to me when I'm reading the code. It looks like you are adding logs to the _global_ namespace. Note that each namespace can have different setting for log appender function, message layout etc. I will try to have a deeper look today and show you what I'm taking about here.

Reply:
So we will probably need to extend all `logger::log_trace` functions in `teal` as none of them uses `namespace = 'teal'`. The same happens in `teal.slice` - none of `logger::log_trace` uses a namespace parameter https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.slice%20log_trace&type=code

Reply:
I think it happens automatically - you can check this comment, where I worked on logging, where each namespace is pre-pended in the logging information

https://github.com/insightsengineering/coredev-tasks/issues/553#issuecomment-2165402290

Reply:
This is what I'm referring to:

``` r
library(logger)

# global
log_threshold(TRACE)
log_layout(layout_glue_generator(format = 'global {level}: {msg}'))
log_appender(appender_stdout)

# namespace: abc
log_threshold(TRACE, namespace = "abc")
log_layout(layout_glue_generator(format = 'ABC {level}: {msg}'), namespace = "abc")
log_appender(appender_stdout)

# test
log_namespaces()
#> [1] "abc"    "global"
```

``` r
log_trace("test")
#> global TRACE: test
```

``` r
log_trace("test", namespace = "abc")
#> ABC TRACE: test
```

<sup>Created on 2024-06-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
On the other hand: https://daroczig.github.io/logger/articles/r_packages.html

> By default, if you are calling the [?log_level](https://daroczig.github.io/logger/reference/log_level.html) function family from an R package after importing from the logger package, then logger will try to auto-guess the calling R package name and use that as the default namespace

> But if auto-guessing is not your style, then feel free to set your custom namespace (eg the name of your package) in all [?log_info](https://daroczig.github.io/logger/reference/log_level.html) etc function calls

It might be actually true that these are automatically added somehow. That will be in line with what you sent. I'm just looking for consistency - to either add `namespace` argument everywhere or nowhere. That's probably a new task. Please just make sure it's matching the current convention - whatever it is

Reply:
The current approach matches the current convention. I've actually seen it multiple times for the last few days so I knew there is no need to append namespace. I think we are good and I would not put `namespace` everywhere as this is another thing to be remembered : )